models/articles.model.js

```diff
@@ -15,3 +15,21 @@ exports.readArticleById = (article_id) => {
     return rows[0];
   });
 };
+//count(comment_id) AS comment_count
+exports.readAllArticles = () => {
+  return db
+    .query(
+      `SELECT articles.author, articles.title, articles.article_id, articles.created_at, articles.topic, articles.votes, articles.article_img_url, count(comments.comment_id) AS comment_count
+        FROM articles 
+        LEFT OUTER JOIN comments ON articles.article_id = comments.article_id 
+        GROUP BY articles.author, articles.title, articles.article_id, articles.created_at, articles.topic, articles.votes, articles.article_img_url
+        ORDER BY articles.created_at DESC;`
+    )
+    .then(({ rows }) => {
+      //convert comment count values from string to Number
+      for (let row of rows) {
+        row.comment_count = Number(row.comment_count);
+      }
```

Comment:
This does work, but its not the ideal way to do this as JS is slow and given a large enough data set this could make a noticeable delay. 

How might you do this using only the sql query instead?